packages/handle/src/index.ts

```diff
@@ -54,21 +35,54 @@ export const normalize = (handle: string): string => {
   return handle.toLowerCase()
 }
 
-export const normalizeAndEnsureValid = (
-  handle: string,
-  availableUserDomains: string[],
-): string => {
+export const normalizeAndEnsureValid = (handle: string): string => {
   const normalized = normalize(handle)
-  ensureValid(normalized, availableUserDomains)
+  ensureValid(normalized)
   return normalized
 }
 
-export const isValid = (
+export const isValid = (handle: string): boolean => {
+  try {
+    ensureValid(handle)
+  } catch (err) {
+    if (err instanceof InvalidHandleError) {
+      return false
+    }
+    throw err
+  }
+  return true
+}
+
+export const ensureServiceConstraints = (
+  handle: string,
+  availableUserDomains: string[],
+  reserved = reservedSubdomains,
+): void => {
+  const supportedDomain = availableUserDomains.find((domain) =>
+    handle.endsWith(domain),
+  )
+  if (!supportedDomain) {
+    throw new UnsupportedDomainError('Not a supported handle domain')
+  }
+  const front = handle.slice(0, handle.length - supportedDomain.length)
+  if (front.indexOf('.') > -1) {
+    throw new InvalidHandleError('Invalid characters in handle')
+  }
+  if (front.length < 3) {
+    throw new InvalidHandleError('Handle too short')
+  }
+  if (reserved[front]) {
+    throw new ReservedHandleError('Reserved handle')
+  }
+}
+
+export const fulfillsServiceConstraints = (
   handle: string,
   availableUserDomains: string[],
+  reserved = reservedSubdomains,
 ): boolean => {
   try {
-    ensureValid(handle, availableUserDomains)
+    ensureServiceConstraints(handle, availableUserDomains, reserved)
   } catch (err) {
     if (
       err instanceof InvalidHandleError ||
```

Comment:
I wonder if we need to add a case here for `UnsupportedDomainError`.
```suggestion
      err instanceof InvalidHandleError ||
      err instanceof UnsupportedDomainError ||
```

Reply:
yup good find 👍